### PR TITLE
Added mongoose_metrics:ensure_subscribed_metric/3

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "mod_global_distrib_utls:ensure_metric/3 does not actually subscribe anything".

Proposed changes include:
* Added mongoose_metrics:ensure_subscribed_metric/3
* Added logging of errors from exometer_report:subscribe
* test

### CODE REVIEW

**Review by JanuszJakubiec — APPROVED** [review]

File: src/metrics/mongoose_metrics.erl [hunk 1]
```diff
@@ -54,6 +55,10 @@
 
 -type use_or_skip() :: use | skip.
 -type hook_name() :: atom().
+-type metric_name() :: atom() | list(atom() | binary()).
+-type short_metric_type() :: spiral | histogram | counter.
```
Comment (JanuszJakubiec) [inline src/metrics/mongoose_metrics.erl:59]:
```suggestion
-type short_metric_type() :: spiral | histogram | counter | gauge.
```

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95096) / small_tests / 42a5dcf3145c7307f89f616d98db0f0a02b3452d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3353/95096/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3353/95096/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-16_14.51.26/lib.mongooseim.logs/run.2021-12-16_14.51.26/suite.log.html)

**Review by JanuszJakubiec — APPROVED** [review]

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95803) / small_tests / 9ef9413ae05fcb1cbfdcb542e4d884a243c8e918
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3353/95803/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3353/95803/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-17_19.27.12/lib.mongooseim.logs/run.2021-12-17_19.27.13/suite.log.html)

**Review by JanuszJakubiec — APPROVED** [review]

**Review by NelsonVides — APPROVED** [review]
Neat 👌🏽

**Pull request merged**